### PR TITLE
Update /instrument flag values

### PR DESCRIPTION
**Pull request by JohnHamby**
This PR is a continuation of #13013 . Unfortunately, I do not have write access to @drognanar 's repository, and so I cannot push commits as part of that PR.

I believe that the most recent commit here addresses all outstanding issues from #13013 , except for my own question about comparing EmitOptions objects that have instrumentation kind sequences with equivalent values in different orders. This question does not require resolution until we have more than one valid instrumentation kind.

The resolution of the MSBuild issue is that instrumentation kinds are specified using properties, not items, named "Instrument". This essentially follows the pattern for NoWarn.

@jcouv @AlekseyTs @cston @gafter should find their issues addressed, with my apologies for not being able to push to the existing PR. Many thanks to @drognanar for doing the vast majority of the work.

@dotnet/roslyn-compiler @dotnet/testimpact @KevinH-MS @ManishJayaswal

### CODE REVIEW

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs [hunk 1]
```diff
@@ -160,10 +160,10 @@ public bool HighEntropyVA
         /// <summary>
         /// Specifies the list of instrumentation kinds to be used during compilation.
         /// </summary>
-        public string InstrumentationKinds
+        public string Instrument
```
Comment (JohnHamby) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: This property is renamed "Instrument" to match the name of the command-line option.

File: src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets [hunk 2]
```diff
@@ -95,7 +95,7 @@
           FileAlignment="$(FileAlignment)"
           GenerateFullPaths="$(GenerateFullPaths)"
           HighEntropyVA="$(HighEntropyVA)"
-          InstrumentationKinds="$(Instrument)"
+          Instrument="$(Instrument)"
```
Comment (JohnHamby) [inline src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: This is renamed "Instrument" to match the name of the command-line option.

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs [hunk 3]
```diff
@@ -755,7 +755,7 @@ internal void AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeeded
             commandLine.AppendPlusOrMinusSwitch("/publicsign", _store, nameof(PublicSign));
             commandLine.AppendSwitchIfNotNull("/runtimemetadataversion:", RuntimeMetadataVersion);
             commandLine.AppendSwitchIfNotNull("/checksumalgorithm:", ChecksumAlgorithm);
-            commandLine.AppendSwitchIfNotNull("/instrument:", InstrumentationKinds);
+            commandLine.AppendSwitchWithSplitting("/instrument:", Instrument, ",", ';', ',');
```
Comment (AlekseyTs) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: What does this do and why can't we simply add the property value?

**Comment by AlekseyTs** [comment]
Do we have targeted test for Equals/GetHashCode of EmitOptions with respect to instrumentation options?

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs (hunk 3 above)
Comment (JohnHamby) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: AppendSwitchWithSplitting divides up a list of options separated by commas or semicolons--it's what is done for NoWarn. Simply adding the property value results in it being quoted when round-tripped through GenerateResponseFileContents.

**Comment by JohnHamby** [comment]
EmitOptionsTests.WithXXX includes a case that does at least rudimentary testing that Equals and GetHashCode work in the presence of instrumentation options.

**Comment by jaredpar** [comment]
@JohnHamby does the testing use the `EqualityUtil` helpers?  If not you should switch to them.  It's pretty simple to use and does exhaustive testing on all of the quirks of .NET equality. 

http://source.roslyn.io/#Roslyn.Compilers.UnitTests/LinePositionTests.cs,15

**Comment by JohnHamby** [comment]
The testing uses exactly the same mechanisms as the existing testing of Equals and GetHashCode for EmitOptions, which do not use EqualUtil helpers. Moving this testing to use EqualUitl helpers is beyond the scope of this change.

**Comment by JohnHamby** [comment]
I have verified that the testing in EmitOptionsTests.WithXXX does exercise value comparisons of instrumentation kinds in EmitOptions.Equals and EmitOptions.GetHashCode. The test, for each kind of emit option, creates two new objects using With*** and compares them for equality and equal hash codes.

**Comment by AlekseyTs** [comment]
LGTM

**Comment by JohnHamby** [comment]
@jcouv signed off on @13013, so the required two signoffs have occurred.

**Comment by JohnHamby** [comment]
Tagging @tannergooding in case we need a new build.

**Comment by tannergooding** [comment]
Build kicked off.

**Pull request merged**